cmd/vulcan-prowler/main.go

```diff
@@ -136,86 +204,168 @@ func main() {
 		}
 
 		logger.Infof("account alias: '%s'", alias)
-
-		report, err := runProwler(ctx, opts)
+		groups, err := groupsFromOpts(opts)
+		if err != nil {
+			return err
+		}
+		r, err := runProwler(ctx, opts.Region, groups)
 		if err != nil {
 			return err
 		}
 
-		addVulnsToState(state, report, alias)
+		var v report.Vulnerability
+		if opts.SecurityLevel == nil {
+			v = CISCompliance
 
-		return nil
+		}
+		if *opts.SecurityLevel == 0 || *opts.SecurityLevel == 1 {
+			v = CISLevel1Compliance
+		}
+
+		if *opts.SecurityLevel == 2 {
+			v = CISLevel2Compliance
+		}
+		fv, err := fillCisLevelVuln(&v, r, alias, opts.SecurityLevel)
+		if err != nil {
+			return err
+		}
+		infov, err := buildInfoVuln(r, alias, opts.SecurityLevel)
+		if err != nil {
+			return err
+		}
+		// if fv == nil it means there were no failed checks so there is no
+		// vuln.
+		if fv != nil {
+			state.AddVulnerabilities(*fv)
+		}
+		state.AddVulnerabilities(infov)
 
+		return nil
 	}
 
 	c := check.NewCheckFromHandler(checkName, run)
 	c.RunAndServe()
 }
 
-func addVulnsToState(state state.State, r *prowlerReport, alias string) {
-	v := CISCompliance
+func groupsFromOpts(opts options) ([]string, error) {
+	// If ROLFP level is specified then it defines the group to use.
```

Comment:
I would avoid referencing ROLFP anywhere, as this is the classification criteria that we have chosen in Adevinta. Externally I believe only the security level should be exposed, as it is independent from how each user of Vulcan actually calculates it.

Reply:
Yes true.